src/hydra-queue-runner/build-remote.cc

```diff
@@ -175,24 +177,305 @@ StorePaths reverseTopoSortPaths(const std::map<StorePath, ValidPathInfo> & paths
     return sorted;
 }
 
+std::pair<Path, AutoCloseFD> openLogFile(const std::string & logDir, const StorePath & drvPath)
```

Comment:
`openFile` and some others aren't exported in a header right? In that case maybe move them into an anonymous namespace:

```c++
namespace {
std::pair<Path, AutoCloseFD> openLogFile(const std::string & logDir, const StorePath & drvPath) {
...
}
}
````

See https://stackoverflow.com/questions/4422507/how-are-unnamed-namespaces-superior-to-the-static-keyword

Reply:
It's in a namespace now, and I marked it `static` in https://github.com/NixOS/hydra/pull/1313